ush/python/pygfs/task/jedi.py

```diff
@@ -0,0 +1,300 @@
+#!/usr/bin/env python3
+
+import os
+from logging import getLogger
+from pprint import pformat
+from typing import List, Dict, Any, Optional
+from jcb import render
+from wxflow import (AttrDict,
+                    chdir, rm_p,
+                    parse_j2yaml, save_as_yaml,
+                    logit,
+                    Task,
+                    Executable,
+                    WorkflowException)
+
+logger = getLogger(__name__.split('.')[-1])
+
+
+class JEDI(Task):
+    """Parent class for JEDI-based tasks
+
+    The JEDI class is the parent class for all
+    Global Data Assimilation System (GDAS) tasks
+    that run JEDI-based applications.
+    """
+
+    def __init__(self, config: Dict[str, Any]) -> None:
+        super().__init__(config)
+
+        _exe_name = os.path.basename(self.task_config.JEDIEXE)
+
+        local_dict = AttrDict(
+            {
+                'jedi_exe': os.path.join(self.task_config.DATA, _exe_name),
+                'jedi_yaml': os.path.join(self.task_config.DATA, os.path.splitext(_exe_name)[0] + '.yaml'),
+                'gdasapp_j2tmpl_dir': os.path.join(self.task_config.PARMgfs, 'gdas')
+            }
+        )
+
+        # Extend task_config with local_dict
+        self.task_config = AttrDict(**self.task_config, **local_dict)
+
+    @logit(logger)
+    def initialize(self) -> None:
+        super().initialize()
+
+        # Generate and write JEDI input YAML file
+        self.get_jedi_config()
+
+        # Link JEDI executable to run directory
+        self.link_jedi_exe()
+
+    @logit(logger)
+    def execute(self, aprun_cmd: str, jedi_args: Optional[List] = None) -> None:
```

Comment:
Running only one executable wouldn't work for the "B-matrix" or "prep obs" jobs for example, thus the suggestion above to rename to something descriptive of the analysis job.